### ISSUE TITLE
True lazy reading to open EMD files

### ISSUE THREAD

**Issue by ZanettaPM** [hyperspy/hyperspy#1966]
Currently lazy reading of EDX spectra in EMD files is not truly lazy. The data is stored in a compressed format in the file. HyperSpy reads the compressed data in memory and uncompresses it lazily. 

This lead to error messages if the number of frame is consequent.

Implementing pure lazy reading could permit to work with big datasets (i.e. cartography with too much frames)

**Comment by francisco-dlp** [comment]
This turned out to be partially a bug, see #2008.

@ZanettaPM, #2008 does not implement true lazy reading, but is it enough for your purposes?

**Comment by ZanettaPM** [comment]
I don't know i have to check this. But the dask array might be too large if the number of frames is high right ?

**Comment by francisco-dlp** [comment]
Yes, it keeps the stream array in memory. I've just opened with 40GB of data (uncompressed) whose stream array only took 1GB of memory. Performing a sum over the whole navigation axes took 20s.

**Comment by ZanettaPM** [comment]
Wow, so it should be enough for me !

**Comment by francisco-dlp** [comment]
Could you tested it?

**Comment by ZanettaPM** [comment]
Actually not now, but i'll let you know. I'll try this asap !

**Comment by francisco-dlp** [comment]
Reopening because #2008 only partially fixes this as it doesn't implement true lazy loading.

**Comment by francisco-dlp** [comment]
The situation should improve further (without fully fixing the issue) with #2012
